frontend/src/components/v2/IconButton/InlineActionIconButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { forwardRef, MouseEventHandler } from "react";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ReactNode } from "@tanstack/react-router";
+import clsx from "clsx";
+
+import { Tooltip } from "../Tooltip";
+import { IconButton } from "./IconButton";
+
+interface InlineActionIconButtonProps {
+  className?: string;
+  onClick?: MouseEventHandler;
+  type?: "submit" | "button";
+  icon: IconProp;
+  hint: ReactNode;
+  isDisabled?: boolean;
+  revealOnGroupHover?: boolean;
+  isHidden?: boolean;
+}
+
+export const InlineActionIconButton = forwardRef<HTMLButtonElement, InlineActionIconButtonProps>(
```

Comment:
I noticed in a lot of components there's usage of `Tooltip`, `IconButton` and `FontAwesomeIcon` to present an action button in a limited space.

I pulled this into a single reusable component. I applied it in the new EditPersonalOverride component, and in a few other components that I changed as part of this work.